### PR TITLE
fix(datetime): disable intersection observer during month update

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

The next/previous buttons in the datetime is offset by 2 months. This is because the intersection observer is pre-maturely firing twice, during the re-render of the navigation to the desired month.

Issue Number: #24712

## What is the new behavior?

The intersection observer is disconnected before the state update/re-render and re-attached after the render and scroll offset has completed. This allows the swipe gesture behavior to operate correctly and fixes the issue for next/previous button to navigate to the desired month (1 month offset).

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/datetime/datetime.tsx [hunk 1]
```diff
@@ -723,6 +723,7 @@ export class Datetime implements ComponentInterface {
           minParts: this.minParts,
           maxParts: this.maxParts
         })) {
+          console.log('this being called');
```
Comment (amandaesmith3) [inline core/src/components/datetime/datetime.tsx:726]: Don't forget to remove this :eyes:

**Review by amandaesmith3 — CHANGES_REQUESTED** [review]

File: core/src/components/datetime/datetime.tsx (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/datetime/datetime.tsx]: Good catch! 💯

**Review by amandaesmith3 — APPROVED** [review]

**Comment by Manishatah** [comment]
Hi Guys,
I am on v 6.1.2 and the issue still persists.
 @amandaejohnston  @sean-perkins

**Comment by amandaejohnston** [comment]
@Manishatah If the issue still occurs on the latest version of Ionic (currently 7.5.1), please open a new issue and include a code reproduction.

**Pull request merged**